openapi/src/main/java/io/helidon/openapi/Serializer.java

```diff
@@ -0,0 +1,349 @@
+                        }
+                    } else {
+                        updatedTuples.add(tuple);
+                    }
+                } else {
+                    updatedTuples.add(tuple);
+                }
+            });
+            node.setValue(updatedTuples);
+        }
+
+        /**
+         * Some SmallRye implementation classes have properties not supported by the implemented interface, so ignore those or
+         * the serialized YAML will contain SmallRye-only properties.
+         *
+         * @param javaBean the bean being serialized
+         * @param property the property being serialized
+         * @return true if the property should be processes; false otherwise
+         */
+        private boolean okToProcess(Object javaBean, Property property) {
+            boolean reject = false;
+            reject |= Parameter.class.isAssignableFrom(javaBean.getClass()) && property.getName().equals("hidden");
+            return !reject;
+        }
+    }
+
+    /**
+     * Suppress the tag output for SmallRye implementation classes so the resulting document can be read into any MP OpenAPI
+     * implementation, not just SmallRye's.
+     */
+    static class TagSuppressingWriter extends PrintWriter {
+
+        private static final Pattern SMALLRYE_IMPL_TAG_PATTERN =
+                Pattern.compile("!!" + Pattern.quote(OpenAPIImpl.class.getPackage().getName()) + ".+$");
+
+        TagSuppressingWriter(Writer out) {
+            super(out);
+        }
+
+        @Override
+        public void write(char[] cbuf, int off, int len) {
+            int effLen = detag(CharBuffer.wrap(cbuf), off, len);
+            if (effLen > 0) {
+                super.write(cbuf, off, effLen);
+            }
+        }
+
+        @Override
+        public void write(String s, int off, int len) {
+            int effLen = detag(s, off, len);
+            if (effLen > 0) {
+                super.write(s, off, effLen);
+            }
+        }
+
+        private int detag(CharSequence cs, int off, int len) {
+            int result = len;
+            Matcher m = SMALLRYE_IMPL_TAG_PATTERN.matcher(cs.subSequence(off, off + len));
+            if (m.matches()) {
+                result = len - (m.end() - m.start());
+            }
+
+            return result;
+        }
+    }
+
+    private static class DelegatingProperty extends Property {
+
+        private final Property delegate;
+
+        private DelegatingProperty(Property delegate, String name) {
+            super(name, delegate.getType());
+            this.delegate = delegate;
+        }
+
+        @Override
+        public Class<?> getType() {
+            return delegate.getType();
+        }
+
+        @Override
+        public Class<?>[] getActualTypeArguments() {
+            return delegate.getActualTypeArguments();
+        }
+
+        @Override
+        public String getName() {
+            return super.getName();
```

Comment:
Surprise: was expecting `delegate`.  Don't know if that's on purpose or not.

Reply:
The whole `DelegatingProperty` class (and the logic that instantiated it) was a holdover from an earlier approach that's no longer used. Removed.